File: Daily/sbgo/Leetcode300-最长递增子序列.go
package sbgo

func lengthOfLIS(nums []int) int {
	n := len(nums)
	if n <= 1 {
		return n
	}
	dp := make([]int, n)
	for i := range dp {
		dp[i] = 1
	}
	maxLis := 0
	for i := 1; i < n; i++ {
		for j := 0; j < i; j++ {
			if nums[i] > nums[j] && dp[i] < dp[j]+1 {
				dp[i] = dp[j] + 1
			}
		}
		maxLis = max(maxLis, dp[i])
	}
	return maxLis
}

File: Leetcode/PY/Leetcode300-最长递增子序列.py
from typing import List
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:  
        n = len(nums)
        if n <= 1:
            return n
        dp = [1] * n
        maxLis = 0
        for i in range(1,n):
            for  j in range(0,i):
                if nums[i] > nums[j] and dp[i] < dp[j] + 1:
                    dp[i] = dp[j] + 1

            maxLis = max(dp[i],maxLis)

        return maxLis            